include/mbedtls/ctr_drbg.h

```diff
@@ -9,13 +9,28 @@
  * Bit Generators</em>.
  *
  * The Mbed TLS implementation of CTR_DRBG uses AES-256 (default) or AES-128
- * as the underlying block cipher.
- *
- *  \warning Using 128-bit keys for CTR_DRBG limits the security of generated
+ * as the underlying block cipher, with a derivation function. The security
+ * strength is:
+ * - 256 bits under the default configuration of the library, with AES-256
+ *   (`MBEDTLS_CTR_DRBG_USE_128_BIT_KEY` not set) and
+ *   with #MBEDTLS_CTR_DRBG_ENTROPY_LEN set to 48 or more.
+ * - 256 bits if AES-256 is used, #MBEDTLS_CTR_DRBG_ENTROPY_LEN is set
+ *   to 32 or more, and the DRBG is initialized with an explicit
+ *   nonce in the \c custom parameter to see mbedtls_ctr_drbg_seed().
+ * - 128 bits if AES-256 is used but #MBEDTLS_CTR_DRBG_ENTROPY_LEN is
+ *   between 24 and 47 and the DRBG is not initialized with an explicit
+ *   nonce (see mbedtls_ctr_drbg_seed()).
+ * - 128 bits if AES-128 is used (`MBEDTLS_CTR_DRBG_USE_128_BIT_KEY` set)
+ *   and #MBEDTLS_CTR_DRBG_ENTROPY_LEN is set to 24 or more (which is
+ *   always the case unless it is explicitly set to a different value
+ *   in `config.h`).
+ *
+ *  \warning Using 128-bit keys for CTR_DRBG or using SHA-256 as the entropy
```

Comment:
I don't think this warning is necessary. The effects of using AES-128 are clearly stated above and intuitive anyway. Also, using SHA-256 won't necessarily limit the security to 128 bits:
- If AES-256 is used with #MBEDTLS_CTR_DRBG_ENTROPY_LEN set to 32 and provides a nonce, then the security strength is 256 bits.
-  If AES-256 is used with #MBEDTLS_CTR_DRBG_ENTROPY_LEN set to 48 (or just to anything above 32), then the application will break because the entropy source fails

Reply:
Removed.